### PR TITLE
Refactor logger's code and remove all emojis from for non-supporting terminals

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/log/logger.go [hunk 1]
```diff
@@ -19,9 +20,13 @@ type LogFormat string
 var Format LogFormat
 
 // Determines whether the terminal is available. This variable should not be accessed directly,
-// but through the 'isTerminalMode' function.
+// but through the 'IsTerminal' function.
 var terminalMode *bool
 
+// Determines whether colors supported. This variable should not be accessed directly,
```
Comment (eyalbe4) [inline utils/log/logger.go:26]: ...colors supported... --> ...colors are supported...

File: utils/log/logger.go [hunk 2]
```diff
@@ -161,24 +194,50 @@ type Log interface {
...
+		supported := true
+
+		if !IsTerminal() ||
+
+			// https://en.wikipedia.org/wiki/Computer_terminal#Dumb_terminals
+			os.Getenv("TERM") == "dumb" ||
+
+			// On Windows WT_SESSION is set by the modern terminal component.
+			// Older terminals have poor support for UTF-8, VT escape codes, etc.
+			runtime.GOOS == "windows" && os.Getenv("WT_SESSION") == "" ||
```
Comment (eyalbe4) [inline utils/log/logger.go:218]: Mayne better to put parentheses around the Windows part in the if statement. Change - 
```runtime.GOOS == "windows" && os.Getenv("WT_SESSION") == "" ||```
into - 
```(runtime.GOOS == "windows" && os.Getenv("WT_SESSION")) == "" ||```

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/log/logger.go [hunk 3]
```diff
@@ -18,13 +19,24 @@ type LogFormat string
 // Used for coloring sections of the log message. For example log.Format.Path("...")
 var Format LogFormat
 
-// Determines whether the terminal is available. This variable should not be accessed directly (besides  tests),,
-// but through the 'isTerminal' function.
-var TerminalMode *bool
+// Determines whether the Stdout is terminal is available. This variable should not be accessed directly,
+// but through the 'IsTerminal' function.
+var StdOutIsTerminal *bool
```
Comment (eyalbe4) [inline utils/log/logger.go:24]: Shound't this variable be private?

File: utils/log/logger.go [hunk 4]
```diff
@@ -18,13 +19,24 @@ type LogFormat string
...
+
+// Determines whether the Stderr is  terminal is available. This variable should not be accessed directly,
+// but through the 'IsTerminal' function.
+var StdErrIsTerminal *bool
```
Comment (eyalbe4) [inline utils/log/logger.go:28]: Shound't this variable be private?

File: utils/log/logger.go [hunk 5]
```diff
@@ -18,13 +19,24 @@ type LogFormat string
 // Used for coloring sections of the log message. For example log.Format.Path("...")
 var Format LogFormat
 
-// Determines whether the terminal is available. This variable should not be accessed directly (besides  tests),,
-// but through the 'isTerminal' function.
-var TerminalMode *bool
+// Determines whether the Stdout is terminal is available. This variable should not be accessed directly,
```
Comment (eyalbe4) [inline utils/log/logger.go:22]: ...is terminal is available...  - Baf grammar. Let's rephrase.

File: utils/log/logger.go [hunk 6]
```diff
@@ -18,13 +19,24 @@ type LogFormat string
...
+// but through the 'IsTerminal' function.
+var StdOutIsTerminal *bool
+
+// Determines whether the Stderr is  terminal is available. This variable should not be accessed directly,
```
Comment (eyalbe4) [inline utils/log/logger.go:26]: ...is terminal is available...  - Baf grammar. Let's rephrase.

File: utils/log/logger.go [hunk 7]
```diff
@@ -74,30 +86,62 @@ func (logger *jfrogLogger) SetLogLevel(LevelEnum LevelType) {
 }
 
 func (logger *jfrogLogger) SetOutputWriter(writer io.Writer) {
-	if writer == nil {
-		writer = os.Stdout
+	if writer != nil {
+		outputWriter = writer
+		// reset outIsTerminal flag
```
Comment (eyalbe4) [inline utils/log/logger.go:91]: reset --> Reset

File: utils/log/logger_test.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+func TestLoggerRemoveEmojis(t *testing.T) {
+	testLoggerWithEmojis(t, false, expectedLogOutputWithoutEmojis)
+}
+
+func TestLoggerLeaveEmojis(t *testing.T) {
+	expected := expectedLogOutputWithEmojis
+	if io.IsWindows() {
+		// should not print emojis on Windows
```
Comment (eyalbe4) [inline utils/log/logger_test.go:17]: should --> Should

File: utils/log/logger_test.go [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+func testLoggerWithEmojis(t *testing.T, mockIsTerminal bool, expected string) {
+	previousLog := Logger
+	// Restore previous logger when the function returns.
+	defer SetLogger(previousLog)
+
+	// Set new logger with output redirection to buffer.
+	buffer := &bytes.Buffer{}
+	SetLogger(NewLogger(DEBUG, buffer))
+	if mockIsTerminal {
+		// Mock logger with true isTerminal flags
```
Comment (eyalbe4) [inline utils/log/logger_test.go:32]: ...with true isTerminal flags --> ...with isTerminal flags set to true

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/log/logger.go (hunk 4 above)
Comment (sverdlov93) [reply utils/log/logger.go:28]: @Or-Geva needs them to be public to manipulate them from outside on his cli tree view tests

**Pull request merged**